src/main/java/org/fenixedu/treasury/services/reports/DocumentPrinter.java

```diff
@@ -54,133 +54,43 @@
 
 import java.util.List;
 
-import org.fenixedu.treasury.domain.Customer;
-import org.fenixedu.treasury.domain.FinantialInstitution;
 import org.fenixedu.treasury.domain.debt.DebtAccount;
 import org.fenixedu.treasury.domain.document.DebitNote;
 import org.fenixedu.treasury.domain.document.FinantialDocument;
-import org.fenixedu.treasury.domain.document.Invoice;
-import org.fenixedu.treasury.domain.document.SettlementNote;
-import org.fenixedu.treasury.domain.document.TreasuryDocumentTemplate;
-import org.fenixedu.treasury.services.reports.dataproviders.CustomerDataProvider;
-import org.fenixedu.treasury.services.reports.dataproviders.DebtAccountDataProvider;
-import org.fenixedu.treasury.services.reports.dataproviders.FinantialInstitutionDataProvider;
-import org.fenixedu.treasury.services.reports.dataproviders.InvoiceDataProvider;
-import org.fenixedu.treasury.services.reports.dataproviders.SettlementNoteDataProvider;
-import org.fenixedu.treasury.services.reports.helpers.DateHelper;
-import org.fenixedu.treasury.services.reports.helpers.EnumerationHelper;
-import org.fenixedu.treasury.services.reports.helpers.LanguageHelper;
-import org.fenixedu.treasury.services.reports.helpers.MoneyHelper;
-import org.fenixedu.treasury.services.reports.helpers.NumbersHelper;
-import org.fenixedu.treasury.services.reports.helpers.StringsHelper;
-
-import com.qubit.terra.docs.core.DocumentGenerator;
-import com.qubit.terra.docs.core.DocumentTemplateEngine;
-import com.qubit.terra.docs.core.IDocumentTemplateService;
 
 public class DocumentPrinter {
-    static {
-        registerService();
-    }
+    private static DocumentPrinterInterface SINGLETON;
 
-    public static final String PDF = DocumentGenerator.PDF;
-    public static final String ODT = DocumentGenerator.ODT;
-
-    private static void registerService() {
-        IDocumentTemplateService service = new DocumentPrinterConfiguration();
-        DocumentTemplateEngine.registerServiceImplementations(service);
+    public static void registerPrinterInterface(DocumentPrinterInterface printer) {
+        SINGLETON = printer;
+        SINGLETON.init();
     }
 
-    private static void registerHelpers(DocumentGenerator generator) {
-        generator.registerHelper("dates", new DateHelper());
-        generator.registerHelper("lang", new LanguageHelper());
-        generator.registerHelper("numbers", new NumbersHelper());
-        generator.registerHelper("enumeration", new EnumerationHelper());
-        generator.registerHelper("strings", new StringsHelper());
-        generator.registerHelper("money", new MoneyHelper());
-    }
+    public static final String PDF = "application/pdf";
+    public static final String ODT = "application/vnd.oasis.opendocument.text";
 
     public static byte[] printDebtAccountPaymentPlan(DebtAccount debtAccount, String outputMimeType) {
-
-        return printDebitNotesPaymentPlan(debtAccount, null, outputMimeType);
+        if (SINGLETON != null) {
```

Comment:
@joaofreitas-qubit we can simplify the code having this part refactored:

`        
 private static <T extends Object> T apply(Function<Xpto, T> function) {
        return Optional.ofNullable(SINGLETON).map(function).orElseThrow(() -> new RuntimeException("Feature not available"));
    }
`

And then you can simply do in the methods:

`
 public static byte[] printDebtAccountPaymentPlan(DebtAccount debtAccount, String outputMimeType) {
   apply(dp -> dp.printDebtAccountPaymentPlan(debtAccount,outputMimeType));
 }
`
Although this can reduce some code, I'm not blocking the PR due to this refactor.